File: LeetCode/45跳跃游戏Ⅱ.java
class Solution {
    public int jump(int[] nums) {
        int ans = 0, cur = 0, next = 0;
        if (nums.length == 1) return 0;
        for (int i = 0; i <= cur && cur < nums.length - 1; i++) {
            next = Math.max(i + nums[i], next);
            if (i == cur) {
                cur = next;
                ans++;
            }
        }
        return ans;
    }
}

File: LeetCode/45跳跃游戏Ⅱ.go
func jump(nums []int) (ans int) {
	if len(nums) == 1 {
		return 0
	}
	ans = 0
	cur, next := 0, 0
	for i := 0; i <= cur && cur < len(nums)-1; i++ {
		if i+nums[i] > next {
			next = i + nums[i]
		}
		if i == cur {
			ans++
			cur = next
		}
	}
	return
}